src/http_server.cc

```diff
@@ -2765,6 +2765,41 @@ HTTPAPIServer::EVBufferToRawInput(
   return nullptr;  // success
 }
 
+struct HeaderSearchPayload {
+  HeaderSearchPayload(
+      const std::string& prefix, TRITONSERVER_InferenceRequest* request)
+      : prefix_(prefix), request_(request), error_(nullptr)
+  {
+  }
+
+  const std::string& prefix_;
+  TRITONSERVER_InferenceRequest* request_;
+  TRITONSERVER_Error* error_;
+};
+
+int
+ForEachHeader(evhtp_header_t* header, void* arg)
+{
+  HeaderSearchPayload* header_search_payload =
+      reinterpret_cast<HeaderSearchPayload*>(arg);
+
+  TRITONSERVER_InferenceRequest* request = header_search_payload->request_;
+  const std::string& prefix = header_search_payload->prefix_;
+
+  // If the prefix is '*' forward all the headers.
+  if (prefix == "*" || (std::string(header->key).find(prefix) == 0)) {
```

Comment:
TBD if `re2` is suitable per offline conversation, but just a nit: `find(prefix)` won't actually enforce that this is a prefix. It may match in the middle of the string. I don't think that's an issue necessarily, just pointing it out if we want to document the behavior explicitly, or just change to startsWith() or similar.

Reply:
C++ doesn't have `startswith`.

> find(prefix) won't actually enforce that this is a prefix.

find by itself won't guarantee that but I made sure that the position returned by `find` is equal to zero (i.e., it is a prefix).

Reply:
Ah I see, misread. Thanks!